izumi-reflect/izumi-reflect/src/main/scala-2/izumi/reflect/ReflectionUtil.scala

```diff
@@ -139,12 +139,47 @@ private[reflect] object ReflectionUtil {
   }
 
   def kindOf(tpe: Universe#Type): Kind = {
-    Kind(tpe.typeParams.map(t => kindOf(t.typeSignature)))
+    Kind(
+      tpe.typeParams.map { param =>
+        val sig = param.typeSignature
+        val bounds = sig match {
+          case tb: Universe#TypeBoundsApi => Some(tb.asInstanceOf[Universe#TypeBounds])
+          case pt: Universe#PolyTypeApi =>
+            pt.resultType match {
+              case tb: Universe#TypeBoundsApi => Some(tb.asInstanceOf[Universe#TypeBounds])
+              case _ => None
+            }
+          case _ => None
+        }
+        val innerParams = sig match {
+          case pt: Universe#PolyTypeApi => pt.typeParams.map(_.asInstanceOf[Universe#Symbol])
+          case _ => Nil
+        }
+        Kind(kindOf(sig).args, bounds, innerParams)
+      },
+      None,
+      tpe.typeParams.map(_.asInstanceOf[Universe#Symbol])
+    )
   }
 
-  final case class Kind(args: List[Kind]) {
+  /** Represents the kind of a type, including bounds.
+    * @param args nested kinds for each type parameter
+    * @param bounds optional type bounds (None for default bounds Nothing..Any)
+    * @param typeParams original type parameter symbols (for substitution during type construction)
+    */
+  final case class Kind(args: List[Kind], bounds: Option[Universe#TypeBounds], typeParams: List[Universe#Symbol] = Nil) {
```

Comment:
@AriajSarkar The last refactoring made this type incoherent. it doesn't make sense to have a separate field for typeParams, when args already represents typeParams. This should work:
```suggestion
  final case class Kind(args: List[Kind], bounds: Option[Universe#TypeBounds], symbol: Universe#Symbol) {
```

Reply:
Makes sense. I see how args already covers that, so separating typeParams is redundant.

Reply:
wait i need to check locally.